R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,23 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  # do not download if it is run in the sparkR shell
+  if (!grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)) {
+    if (!nzchar(master) || is_master_local(master)) {
```

Comment:
shouldn't we also fail if master != local but SPARK_HOME is not defined or spark jar is not in SPARK_HOME?